test/Microsoft.AspNet.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -42,11 +42,18 @@ private class Address
                 ParameterType = typeof(Person)
             };
 
-            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
-            var httpContext = operationContext.HttpContext;
-
-            ConfigureHttpRequest(httpContext.Request, string.Empty);
-            var modelState = new ModelStateDictionary();
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(
+              request =>
+              {
+                  request.Body = new MemoryStream(Encoding.UTF8.GetBytes("{ \"Id\":1234 }"));
+                  request.ContentType = "application/json";
+              });
+
+            var actionContext = operationContext
+                .HttpContext
+                .RequestServices
+                .GetRequiredService<IScopedInstance<ActionContext>>().Value;
+            var modelState = actionContext.ModelState;
```

Comment:
>Really not a fan of this showing up in tests. What's the need for this? Is this because formatters use the IScopedInstance<ActionContext> to access model state?

Yes- And yes that is a bug that we intend to fix ( Json Formatter would not add errors using model state ). This would be cleaned up when we do that .. 

>As an alternative can we just have the tests initialize things in the order they need to go?

I would like the tests to avoid handling multiple contexts, personally i like the delegate approach.